src/Microsoft.AspNetCore.Mvc.Analyzers/CodeAnalysisExtensions.cs

```diff
@@ -11,6 +11,14 @@ internal static class CodeAnalysisExtensions
     {
         public static bool HasAttribute(this ITypeSymbol typeSymbol, ITypeSymbol attribute, bool inherit)
         {
+            Debug.Assert(typeSymbol != null);
```

Comment:
Are these extensions common enough to go into a common repo that could be used by analyzers written in other repos?

Reply:
We can consider this if there's an ask. Most of the logging analyzers look at strings so they wouldn't benefit from this.